### ISSUE TITLE
Roslyn doesn't respect flow annotations when checking ref argumengs

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#42722]
**Version Used**: 

```
Branch master (10 Mar 2020)
Latest commit 441c154 by msftbot[bot]:
Merge pull request #42303 from CyrusNajmabadi/useCompoundParens

Support parenthesized expressions in use-compound-operator.
```

**Steps to Reproduce**:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

class C
{
  void M([MaybeNull] ref string s)
  {
    Test(ref s);
  }
  
  void Test(ref string? s) => s = null;
}
```

**Expected Behavior**:
No warnings. It's OK to assign `s` with null e.g. `s = null` doesn't produce a warning.

**Actual Behavior**:
`warning CS8601: Possible null reference assignment.` is reported for `ref s` argument
